### ISSUE TITLE
TensorFlow Object Detection training error

### ISSUE THREAD

**Issue by mkaur013** [tensorflow/models#10565]
Hello, 

I am training a object detection model 
TensorFlow = 2.7.0
If I use TensorFlow = 2.8.0, then i get error "No module named 'tensorflow.python.keras.layers.preprocessing'"
Got below error:

Operation defined at: (most recent call last)
>>>   File "C:\Users\User\anaconda3\lib\threading.py", line 890, in _bootstrap
>>>     self._bootstrap_inner()
>>>
>>>   File "C:\Users\User\anaconda3\lib\threading.py", line 926, in _bootstrap_inner
>>>     self.run()
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\object_detection\model_lib_v2.py", line 170, in _dummy_computation_fn
>>>     return _compute_losses_and_predictions_dicts(model, features, labels,
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\object_detection\model_lib_v2.py", line 123, in _compute_losses_and_predictions_dicts
>>>     prediction_dict = model.predict(
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\object_detection\meta_architectures\ssd_meta_arch.py", line 569, in predict
>>>     if self._feature_extractor.is_keras_model:
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\object_detection\meta_architectures\ssd_meta_arch.py", line 570, in predict
>>>     feature_maps = self._feature_extractor(preprocessed_inputs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\utils\traceback_utils.py", line 64, in error_handler
>>>     return fn(*args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\engine\base_layer.py", line 1083, in __call__
>>>     outputs = call_fn(inputs, *args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\utils\traceback_utils.py", line 92, in error_handler
>>>     return fn(*args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\object_detection\meta_architectures\ssd_meta_arch.py", line 251, in call
>>>     return self._extract_features(inputs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\object_detection\models\ssd_mobilenet_v2_fpn_keras_feature_extractor.py", line 219, in _extract_features
>>>     image_features = self.classification_backbone(
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\utils\traceback_utils.py", line 64, in error_handler
>>>     return fn(*args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\engine\base_layer.py", line 1083, in __call__
>>>     outputs = call_fn(inputs, *args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\utils\traceback_utils.py", line 92, in error_handler
>>>     return fn(*args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\engine\functional.py", line 452, in call
>>>     inputs, training=training, mask=mask)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\engine\functional.py", line 589, in _run_internal_graph
>>>     outputs = node.layer(*args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\utils\traceback_utils.py", line 64, in error_handler
>>>     return fn(*args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\engine\base_layer.py", line 1083, in __call__
>>>     outputs = call_fn(inputs, *args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\utils\traceback_utils.py", line 92, in error_handler
>>>     return fn(*args, **kwargs)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\layers\convolutional.py", line 246, in call
>>>     outputs = self.convolution_op(inputs, self.kernel)
>>>
>>>   File "C:\Users\User\Desktop\ACoBot\Code\tfod\lib\site-packages\keras\layers\convolutional.py", line 238, in convolution_op
>>>     name=self.__class__.__name__)
>>>

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by mkaur013** [comment]
Sorry, I submitted issue first time. 

I am using Jupyter notebook
I am following https://github.com/nicknochnack/TFODCourse/blob/main/2.%20Training%20and%20Detection.ipynb for training  the objects

What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution Windows 10 Pro 
TensorFlow installed from
TensorFlow version 2.8.0
Bazel version
CUDA/cuDNN version 11.2/8.1 (Tried latest 11.6/8.3, also tried training without using gpu)
GPU model and memory NVIDIA Quadro RTX 6000, total graphic memory: 40868 MB, video memory: 24576 MB
Exact command to reproduce "command = "python {} --model_dir={} --pipeline_config_path={} --num_train_steps=2000".format(TRAINING_SCRIPT, paths['CHECKPOINT_PATH'],files['PIPELINE_CONFIG'])"

**Comment by jvishnuvardhan** [comment]
@mkaur013 Is this still an issue for you?

I tried with `TF2.9.1` and I don't that specific error. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/06dc70154d8653bd6e3473a779192822/2-training-and-detection.ipynb).

Please close the issue if this was already resolved for you. Thanks!
